lms/djangoapps/support/static/support/jsx/entitlements/components/EntitlementForm/index.jsx

```diff
@@ -0,0 +1,146 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { Button, InputSelect, InputText, TextArea } from '@edx/paragon';
+
+
+class EntitlementForm extends React.Component {
+  constructor(props) {
+    super(props);
+
+    const isReissue = props.entitlement !== null && props.entitlement !== undefined;
+
+    this.state = {
```

Comment:
Instead of the ternaries, consider

```javascript
const {
  courseUuid,
  mode,
  user,
} = props.entitlement;

let state = { 
  isReissue, 
  courseUuid: '',
  mode: '',
  username: '',
  comments: 'Add any additional comments here' 
};

if (isReissue) {
  state = { ...state, courseUuid, mode, username };
}
```

this.setState(state);

Reply:
only problem with this is when props.entitlement is null, though I suppose if I passed in some default for that it would be fine?